packages/schema/src/api-key/index.ts

```diff
@@ -0,0 +1,61 @@
+import { z } from 'zod'
+import { expiresAfterEnum, authorityEnum } from '@/enums'
+import { PageRequestSchema } from '@/pagination'
+
+export const ApiKeySchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  slug: z.string(),
+  value: z.string(),
+  expiresAt: z.string().datetime().nullable(),
+  createdAt: z.string().datetime(),
+  updatedAt: z.string().datetime(),
+  authorities: z.array(authorityEnum),
+  userId: z.string()
+})
+
+export const CreateApiKeyRequestSchema = z.object({
+  name: ApiKeySchema.shape.name,
+  expiresAfter: expiresAfterEnum.optional(),
```

Comment:
Derive this from the parent shape and make it optional

Reply:
Doing this, however the dto for create end point had only these properties.